### PR TITLE
flameshot: init at 0.5.0

### PR DESCRIPTION
**Pull request by scode**
###### Motivation for this change

flameshot is very useful, let's add it!

###### Things done

(tested on Ubuntu)

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/tools/misc/flameshot/default.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  version = "0.5.0";
+
+  nativeBuildInputs = [ qmake qttools ];
+  buildInputs = [ qtbase ];
+
+  qmakeFlags = [
+    # flameshot.pro assumes qmake is being run in a git checkout and uses it
+    # to determine the version being built. Let's replace that.
+    "VERSION=${version}"
+    "PREFIX=/"
```
Comment (scode) [inline pkgs/tools/misc/flameshot/default.nix]: Cargo culted. Not sure if this is idiomatic.

File: pkgs/tools/misc/flameshot/default.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  ];
+  patchPhase = ''
+    sed -i 's/VERSION =/#VERSION =/g' flameshot.pro
+    sed -i 's,USRPATH = /usr/local,USRPATH = /,g' flameshot.pro
+  '';
+
+  installFlags = [ "INSTALL_ROOT=$(out)" ];
```
Comment (scode) [inline pkgs/tools/misc/flameshot/default.nix]: Cargo culted. Not sure if this is idiomatic.

File: pkgs/tools/misc/flameshot/default.nix [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+  qmakeFlags = [
+    # flameshot.pro assumes qmake is being run in a git checkout and uses it
+    # to determine the version being built. Let's replace that.
+    "VERSION=${version}"
+    "PREFIX=/"
+  ];
+  patchPhase = ''
+    sed -i 's/VERSION =/#VERSION =/g' flameshot.pro
+    sed -i 's,USRPATH = /usr/local,USRPATH = /,g' flameshot.pro
```
Comment (scode) [inline pkgs/tools/misc/flameshot/default.nix]: This feels ugly, but not sure if there's a cleaner way. `flameshot.pro` contains hard-coded `USRPATH` that can take one of two values depending on whether `CONFIG` contains `packaging`.

File: pkgs/tools/misc/flameshot/default.nix [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    sha256 = "1fy4il7rdj294l9cs642hx23bry25j9phn37274r2b87hwzy1rrv";
+  };
+
+  enableParallelBuilding = true;
+
+  meta = {
+    description = "Powerful yet simple to use screenshot software";
+    homepage = https://github.com/lupoDharkael/flameshot;
+    license = stdenv.lib.licenses.gpl3;
+    platforms = stdenv.lib.platforms.all;
```
Comment (scode) [inline pkgs/tools/misc/flameshot/default.nix]: As indicated in the PR I have only tested this on Ubuntu. From grepping around, i get the impression that `.all` would be appropriate in this case anyway.

File: pkgs/tools/misc/flameshot/default.nix (hunk 3 above)
Comment (scode) [reply pkgs/tools/misc/flameshot/default.nix]: For convenience, here's flameshot.pro: https://github.com/lupoDharkael/flameshot/blob/v0.5.0/flameshot.pro

**Comment by grahamc** [comment]
Looks pretty good, can you:

1. reword the first commit to remove the `(` and `)`s 
2. reword the second commit to say "maintainers: add scode"

@GrahamcOfBorg build flameshot

**Comment by scode** [comment]
@grahamc Done. Apologies. I'm so used to a squash based workflow that I just pushed another commit without thinking. Also fixed the commit msg.

**Comment by scode** [comment]
Not sure why it's failing *like this* on Darwin since I'd expect the compilers to be the same.

**Comment by Mic92** [comment]
`darwin` uses `llvm` while on linux we have gcc.

**Comment by scode** [comment]
Reproed locally with clang. I'll submit a PR upstream.

However, curious what the general policy is for things like these. Other than of course preferring to make upstream portable, would the strategy be to let it go in and not work on darwin? Or change the platform to just include linux? Or go ahead and provide patches in nix?

**Comment by scode** [comment]
https://github.com/lupoDharkael/flameshot/pull/101 for the upstream fix

**Comment by scode** [comment]
FWIW, upstream is merged (but there's no release containing it yet).

Thoughts on best path forward?

**Pull request merged**